include/roboteam_ai/world/Ball.h

```diff
@@ -15,41 +11,130 @@ namespace world {
+         * otherwise */
+        bool visibleByAnyCamera = false;
+        // The position where the ball is expected to end (lay still) after following his path.
+        Vector2 expectedBallEndPosition = Vector2();
+        // The velocity but then adjusted to determine a more realistic expectedBallEndPosition
+        Vector2 filteredVelocity = Vector2();
+
+        /**
+         * When the position of the ball is not yet initialized, but was previously near to a robot then initialize
+         * the position of the ball to a position close to that robot.
+         */
+        void initBallAtRobotPosition(const Ball &oldBall, const WorldData &worldData);
+
+        /**
+         * Adds a moving average over the Kalman filter to make the ball-velocity more stable (this method could be
+         * moved to rtt_world).
+        */
+        void filterBallVelocity(Ball &oldBall, const WorldData &worldData);
+
+        /**
+         * Update the expected position where the ball will end (lay still) after following his path.
+         */
+        void updateExpectedBallEndPosition(const Ball &oldBall, const WorldData &worldData);
+
+        /**
+         * When the position of the ball is close to a robot and not visible by camera then update the ball according
+         * to the movement and rotation of that robot (which happens during dribbling).
+         */
+        void updateBallAtRobotPosition(const Ball &oldBall, const WorldData &worldData);
 
     public:
+        static bool exists;
+
+        /**
+         * Create an uninitialized ball (where the expected position and expected velocity are not set yet).
+         */
         Ball();
-        // Ball(const Ball &copy) = default;
+
+        /**
+         * Create a Ball object with the current data about the ball.
+         * @param copy The current data about the ball
+         */
         explicit Ball(const roboteam_proto::WorldBall &copy);
+
+        /**
+         * Update the expected end (lay still) position of the ball after the ball has followed his path. Moreover
+         * this method updates the expected position of the ball if the ball is close to a robot (which happens during
+         * dribbling).
+         * @param oldBall The old state of the ball
+         * @param worldData The current data about the world
+         */
         void updateBall(const BallPtr &oldBall, const WorldData &worldData);
 
-        const Vector2 &getBallStillPosition() const;
-        Vector2 pos = Vector2();
-        Vector2 vel = Vector2();
-        Vector2 acc = Vector2();
-        Vector2 filteredVel = Vector2();
-        double spin = 0.0;
-        static bool exists;
-        bool visible = false;
-        void filterBallVelocity(Ball &oldBall, const WorldData &worldData);
+        /**
+         * Get the expected position of the ball.
+         * @return A vector which represents this position.
+         */
+        const Vector2 &getPos() const;
```

Comment:
I think when something is performing the functionality of getting a variable, the method name should be getThing(). This is helpful because if you want to "get" a property, then you can just type "get" and you autocomplete will show you all of the get methods in once place.

Reply:
Agree with @RobotJesse on this one.

Reply:
I also agree with the getter naming. However, why not use the full name (getPosition, getVelocity)?